src/Domain/LeanCode.CQRS.Execution/DevResponseLoggerElement.cs

```diff
@@ -0,0 +1,38 @@
+using LeanCode.Pipelines;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+
+namespace LeanCode.CQRS.Execution
+{
+    public class DevResponseLoggerElement<AppContext, TObj, TRes> : IPipelineElement<AppContext, TObj, TRes>
+        where AppContext : notnull, IPipelineContext
+    {
+        private readonly ILogger logger;
+        private readonly IHostEnvironment environment;
+
+        public DevResponseLoggerElement(IHostEnvironment env)
+        {
+            environment = env;
+            logger = Log.ForContext<DevResponseLoggerElement<AppContext, TObj, TRes>>();
+        }
+
+        public DevResponseLoggerElement(IHostEnvironment env, ILogger logger)
+        {
+            environment = env;
+            this.logger = logger;
+        }
+
+        public Task<TRes> ExecuteAsync(
+            AppContext ctx,
+            TObj input,
+            Func<AppContext, TObj, Task<TRes>> next)
+        {
+            if (!environment.IsProduction())
+            {
+                logger.Information("Request executed with response {@Response}", input);
```

Comment:
You're logging the wrong thing - you should not log the input, but the response.

Reply:
As I understand it, in this situation input is the response

Reply:
The query handler will be executed when called `next`, not _before_. Elements are run _before_ or _after_ execution, where `before` is before calling `next`, and `after` is after calling `next`. `next` has the result, `input` is _the input_, so a query object.